src/Microsoft.VisualStudio.Composition.Analyzers/VSMEF007DuplicateImportAnalyzer.cs

```diff
@@ -0,0 +1,167 @@
+        title: Strings.VSMEF007_Title,
+        messageFormat: Strings.VSMEF007_MessageFormat,
+        helpLinkUri: Utils.GetHelpLink(Id),
+        category: "Usage",
+        defaultSeverity: DiagnosticSeverity.Warning,
+        isEnabledByDefault: true);
+
+    /// <inheritdoc/>
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Descriptor);
+
+    /// <inheritdoc/>
+    public override void Initialize(AnalysisContext context)
+    {
+        context.EnableConcurrentExecution();
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            // Only scan further if the compilation references the assemblies that define the attributes we'll be looking for.
+            if (Utils.ReferencesMefAttributes(context.Compilation))
+            {
+                context.RegisterSymbolAction(AnalyzeType, SymbolKind.NamedType);
+            }
+        });
+    }
+
+    private static void AnalyzeType(SymbolAnalysisContext context)
+    {
+        var namedType = (INamedTypeSymbol)context.Symbol;
+
+        // Skip interfaces, enums, and other non-class types
+        if (namedType.TypeKind is not (TypeKind.Class or TypeKind.Struct))
+        {
+            return;
+        }
+
+        var importContracts = new List<ImportContract>();
+
+        // Collect all imports from properties
+        foreach (var member in namedType.GetMembers())
+        {
+            if (member is IPropertySymbol property)
+            {
+                var importAttribute = Utils.GetImportAttribute(property.GetAttributes());
+                if (importAttribute is not null)
+                {
+                    var contract = GetImportContract(importAttribute, property.Type);
+                    if (contract is not null)
+                    {
+                        importContracts.Add(new ImportContract(contract, property.Name, property.Locations[0]));
+                    }
+                }
+            }
+        }
+
+        // Collect all imports from constructor parameters
+        foreach (var member in namedType.GetMembers())
+        {
+            if (member is IMethodSymbol method && method.MethodKind == MethodKind.Constructor)
+            {
+                // Check if this constructor has [ImportingConstructor] or if it's the only constructor
+                bool isImportingConstructor = Utils.HasImportingConstructorAttribute(method) ||
+                    (namedType.Constructors.Length == 1 && Utils.HasInstanceExports(namedType));
+
+                if (isImportingConstructor)
+                {
+                    foreach (var parameter in method.Parameters)
+                    {
+                        var importAttribute = Utils.GetImportAttribute(parameter.GetAttributes());
+                        if (importAttribute is not null)
```

Comment:
An importing constructor parameter is still an import even if it lacks this attribute. Is this a blindspot in the analyzer?

Reply:
Yes, fixed.